### PR TITLE
Settings saving implementation

### PR DESCRIPTION
**Pull request by NancokPS2**
- The options menu now saves settings to a file whenever a submenu or the menu itself is closed  
- Root loads all local settings during _ready  
- All mappings for actions that can be remapped from the options menu are saved
- Volume is also saved
  
To test: change any settings (volume or button mappings), close the menu and restart the game, the settings should remain as they where in the previous session.  
The first time the game is launched, a warning will popup about there not being any saved data present, this is normal.
  
Sample of a modified savefile's conents:  
  
```
[SETTINGS]

last_launch_date="2023-10-11"
volume=0.08
controlMappings={
"j_toggle_bag": [Object(InputEventKey,"resource_local_to_scene":false,"resource_name":"","device":0,"window_id":0,"alt_pressed":false,"shift_pressed":false,"ctrl_pressed":false,"meta_pressed":false,"pressed":true,"keycode":70,"physical_keycode":70,"key_label":70,"unicode":102,"echo":false,"script":null)
],
"j_toggle_equipment": [Object(InputEventKey,"resource_local_to_scene":false,"resource_name":"","device":0,"window_id":0,"alt_pressed":false,"shift_pressed":false,"ctrl_pressed":false,"meta_pressed":false,"pressed":true,"keycode":67,"physical_keycode":67,"key_label":67,"unicode":99,"echo":false,"script":null)
]
}
```

### CODE REVIEW

**Comment by jonathaneeckhout** [comment]
Tested it:

- Configure the volume to a certain level, stop the game, restart -> it's back to the level configured level -> test ok
- I remapped the bag to "t" -> restarted the client -> "t" was not working, it was "b" again -> test nok

**Comment by NancokPS2** [comment]
I can't reproduce the bug, did you close the game before closing the options menu?  
Also, once you remap something, check the saved file with a text editor.

**Comment by jonathaneeckhout** [comment]
The content of the file is:

[SETTINGS]

thisThing="it exists"
volume=0.22

However I did reset the controls for toggle bag to T instead of B.

**Comment by NancokPS2** [comment]
And there's no errors either?  
I'm not sure what else to try. I already pulled all changes from main to test them in the branch

**Comment by jonathaneeckhout** [comment]
I will have a look once I got time to help debug it.

File: scenes/ui/options_menu/input_remapping.gd [hunk 1]
```diff
@@ -40,6 +40,8 @@ func _ready() -> void:
 	remap_aborted.connect( temporary_text_timed.bind(Messages.ABORTED, tempTextDuration) )
 	remap_successful.connect( temporary_text_timed.bind(Messages.SUCCESSFUL, tempTextDuration) )
 	
+	tree_exiting.connect( save_mappings )
```
Comment (jonathaneeckhout) [inline scenes/ui/options_menu/input_remapping.gd:43]: So I've found that this is only called when you press the "close" button.

I can confirm it works if you set a mapping and close it via this button.

But, a user can also close the menu by pressing the "esc" key. If you press the esc key, this signal is not triggered, which causes the mapping to not be stored

Comment (jonathaneeckhout) [reply scenes/ui/options_menu/input_remapping.gd:43]: Otherweird behavior spotted:
if you press esc to close the menu the tree_exiting is not triggered, but if you reopen the menu and press the option menu, it is triggered.

Comment (NancokPS2) [reply scenes/ui/options_menu/input_remapping.gd]: Fixed, now it saves from the options_menu script instead of from the mapping menu.

**Comment by NancokPS2** [comment]
> Tested again, it now saves every time you press the close button, however when pressing the "esc" keyboard key it does not.
> 
> I was wondering if it would make it clearer for the end user to explicitly add a "save" button? Only when the save button is pressed, the new keybinds take effect and they are stored to the persistent file.

That is odd, it uses the exact same saving logic that it uses for closing the menu (in fact, the close_menu() function is what calls the saving), there should be no possible way to close the menu without having the file get saved.  
I will double check later.

**Comment by jonathaneeckhout** [comment]
works like a charm, merging.

Thanks for your contribution.

**Pull request merged**